### PR TITLE
Upgrade to astro 3, add view transitions/CSR

### PR DESCRIPTION
**Pull request by Scott2bReal**
## Description

Upgrades the app to Astro 3 (all deps now at `latest`), `node-version` to `20.5.0`, and adds a few features:

* Optimized images - the few images on the site are now using Astro's [stable `Image` component](https://docs.astro.build/en/guides/images/#image--astroassets). Further optimizations are on the table...
* View transitions - uses Astro's [new `ViewTransitions` component](https://docs.astro.build/en/guides/view-transitions/) to enable client side routing (and fixes some resulting JS bugs)
* Removes element causing sticky points in page scrolling

Got a little carried away with upgrades while working on this and added some project-wide scripts for linting and formatting, so this PR is a bit noisy. Will likely add github actions to run these as checks on PRs in the future.

## Testing
* View on Chrome, navigate between home and /contact to see new page transitions (should fade in/out)
* Navigate on Firefox to make sure fallback looks good (view transitions are not implemented in Firefox yet)
* Confirm animations on homepage fire on initial load & after navigation via client side router

### CODE REVIEW

File: astro/src/pages/index.astro [hunk 1]
```diff
@@ -48,35 +50,36 @@ const builder = sanityImageUrl(banner.image)
...
-      >
-        <h2 class="p-4 text-3xl">VIDEO</h2>
-        <RandomSnacksVid snacksVids={snacksVids} client:idle />
-      </section>
-
-      <!-- Band info -->
-      <section id="band">
-        <BandInfo />
-      </section>
+    <div class="bg-none pt-[30vh] sm:pt-[100vh]">
```
Comment (Scott2bReal) [inline astro/src/pages/index.astro:53]: This diff got a little messy due to formatting changes, so it's kind of hard to tell at a glance what changed. 

Previously we had a `100vh` div with no background that stacked on top of the page content to expose the `position: fixed` hero image. That was causing some weird page scroll stickiness here and there, so I opted to take that out and add `100vh` padding on top of the page content, and add a container inside of that with the black background

File: astro/src/components/BandInfo.astro [hunk 2]
```diff
@@ -120,6 +121,5 @@ const builder = sanityImageUrl(groupPhoto.image)
   }
 
   animateOnScroll()
-
-  // fadeIn();
+  document.addEventListener("astro:page-load", animateOnScroll)
```
Comment (Scott2bReal) [inline astro/src/components/BandInfo.astro:123]: With the addition of client-side routing to the app, a lot of this vanilla JS animation wasn't working. Needed to add a listener for the `astro:page-load` event that fires after successful client-side navigation.

[More context](https://docs.astro.build/en/guides/view-transitions/#script-behavior-during-page-navigation)

File: astro/src/components/BandInfo.astro [hunk 3]
```diff
@@ -12,21 +13,21 @@ const roster = rosterResult.result as BandMember[]
...
   >
     WE ARE NASTY SNACKS
   </h2>
   <div
-    class="max-w[850px] z-10 flex translate-y-6 flex-col px-4 text-center opacity-0 transition duration-1000 ease-in-out md:flex-row md:justify-center"
+    class="z-10 flex max-w-[850px] translate-y-6 flex-col px-4 text-center opacity-0 transition duration-1000 ease-in-out md:flex-row md:justify-center"
     id="bandPic"
   >
-    <img
+    <Image
```
Comment (Scott2bReal) [inline astro/src/components/BandInfo.astro:27]: I wanted to use the `Image` component

File: astro/src/utils/helpers.ts
Comment (Caleb-Mitchell) [inline astro/src/utils/helpers.ts:1]: Commenting for practice, phew I'm glad these are double quotes for consistency.

**Review by Caleb-Mitchell — APPROVED** [review]
Especially given that you walked me through these changes personally, this all looks good to me.

**Pull request merged**